src/Dfc.CourseDirectory.Web/Controllers/CopyCourse/CopyCourseRunController.cs

```diff
@@ -262,243 +219,355 @@ private async Task<SelectVenueModel> GetVenuesByUkprn(int ukprn)
             return selectVenue;
         }
 
-
         [HttpGet]
         [Authorize]
-        public async Task<IActionResult> Index(string learnAimRef, string notionalNVQLevelv2, string awardOrgCode, string learnAimRefTitle, string learnAimRefTypeDesc,  Guid? courseId, Guid courseRunId, PublishMode publishMode)
+        public async Task<IActionResult> Index(Guid? courseId, Guid? courseRunId)
         {
-            int? UKPRN;
+            if (!_session.GetInt32("UKPRN").HasValue)
+            {
+                return RedirectToAction("Index", "Home", new { errmsg = "Please select a Provider." });
+            }
+
+            var ukprn = _session.GetInt32("UKPRN").Value;
+
+            var savedModel = _session.GetObject<CopyCourseRunSaveViewModel>(CopyCourseRunSaveViewModelSessionKey);
 
-            if (_session.GetInt32("UKPRN") != null)
+            if (courseId.HasValue && courseRunId.HasValue)
+            {
+                if (savedModel != null)
+                {
+                    if (savedModel.CourseId != courseId || savedModel.CourseRunId != courseRunId)
+                    {
+                        _session.Remove(CopyCourseRunSaveViewModelSessionKey);
+                        savedModel = null;
+                    }
+                    else
```

Comment:
Isn't this `else` block redundant?

Reply:
No, as you only want to execute that code when the previous condition fails. These conditions are quite confusing so I'm going to invert this if to make it clearer.